extensions/spring-cache/deployment/src/main/java/io/quarkus/spring/cache/SpringCacheProcessor.java

```diff
@@ -0,0 +1,93 @@
+package io.quarkus.spring.cache;
+
+import static io.quarkus.spring.cache.SpringCacheUtil.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.CachePut;
+import org.springframework.cache.annotation.Cacheable;
+
+import io.quarkus.arc.deployment.AnnotationsTransformerBuildItem;
+import io.quarkus.cache.deployment.AdditionalCacheNameBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+
+public class SpringCacheProcessor {
+
+    static final DotName CACHEABLE = DotName.createSimple(Cacheable.class.getName());
+    static final DotName CACHE_PUT = DotName.createSimple(CachePut.class.getName());
+    static final DotName CACHE_EVICT = DotName.createSimple(CacheEvict.class.getName());
+
+    private static final List<DotName> CACHE_ANNOTATIONS = Collections
+            .unmodifiableList(Arrays.asList(CACHEABLE, CACHE_PUT, CACHE_EVICT));
+
+    // some of these restrictions can probably lifted by us doing additional work on caching after https://github.com/quarkusio/quarkus/pull/8631 lands
+    private static final Set<String> CURRENTLY_UNSUPPORTED_ANNOTATION_VALUES = Collections
+            .unmodifiableSet(new HashSet<>(Arrays.asList("key", "keyGenerator", "cacheManager", "cacheResolver", "condition",
+                    "unless", "sync", "beforeInvocation")));
```

Comment:
Yes, I plan to add support for it. It shouldn't be too tough, but it will require more control over the interceptors than we currently have. We'll need something akin to what Spring Security uses.